atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/queue/KvsSweepQueueProgress.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2018 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.sweep.queue;
+
+import java.util.Map;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Iterables;
+import com.palantir.atlasdb.keyvalue.api.Cell;
+import com.palantir.atlasdb.keyvalue.api.CheckAndSetException;
+import com.palantir.atlasdb.keyvalue.api.CheckAndSetRequest;
+import com.palantir.atlasdb.keyvalue.api.KeyValueService;
+import com.palantir.atlasdb.keyvalue.api.TableReference;
+import com.palantir.atlasdb.keyvalue.api.Value;
+import com.palantir.atlasdb.schema.generated.SweepShardProgressTable;
+import com.palantir.atlasdb.schema.generated.TargetedSweepTableFactory;
+import com.palantir.util.PersistableBoolean;
+
+public class KvsSweepQueueProgress {
+    private static final TableReference TABLE_REF = TargetedSweepTableFactory.of()
+            .getSweepShardProgressTable(null).getTableRef();
+    private static final int NUMBER_OF_SHARDS_INDEX = 1024;
+    private static final long CAS_TIMESTAMP = 1L;
+
+    private final KeyValueService kvs;
+
+    public KvsSweepQueueProgress(KeyValueService kvs) {
+        this.kvs = kvs;
+    }
+
+    public long numberOfShards() {
+        return getOrInitializeTo(NUMBER_OF_SHARDS_INDEX, true, 1L);
```

Comment:
Strong suggestion: use an enum rather than bool, for readability and to reduce mental mapping.